### PR TITLE
[AC-96: Feature] admin panel appearance manage account profile

### PR DESCRIPTION
**Pull request by pycxu**
### Provide description as to what you have done in this pr.
Create `updateProfile` mutation to update user `displayName` and `bio`
### Provide screenshots or videos that the function is working.

https://user-images.githubusercontent.com/43846910/178965443-29f8a781-fed2-41a0-acf9-b7c357dab2e8.mov

### Provide screenshots of no warning, error in console.
Screen Shot 2022-07-14 at 6 32 53 pm

#### That the feature you have worked on has its...

- [ ] relevant information updated in `README.MD`
- [ ] swagger doc created/updated so that other developers can read the related documentation.
- [x] logging included/updated so that we can debug errors when the app is live in the UAT environment.
- [ ] unit tests created/updated so that we can integrate this new release and sleep without worries at night/day.

#### When you finished, you will want to make sure you have...

- [x] No merge conflict with `develop` branch.
- [x] Check PR title is correct, (i.e `[AC-xx: Feature] Create pull request template`).
- [x] Check your app will run without crashes.
- [x] Check if there's hard coded values.
- [x] No `//eslint-disable-next-line no-unused-vars`, unless otherwise specified.
- [x] No unnecessary spaces between lines of code.
- [x] Check you are using `camelCase && PascalCase` correctly.
- [x] Check you named your code properly.
- [x] Remove unused comments.

### CODE REVIEW

File: src/resource/user/user.service.ts [hunk 1]
```diff
@@ -134,4 +134,24 @@ export class UserService {
...
     });
   }
+
+  async updateProfile({ username, displayName, bio }: UpdateProfileInput) {
+    const updatedData = {
+      displayName,
+      bio,
+    };
+    const updatedUser = await this.prisma.user.update({
+      where: { username },
```
Comment (NorrisWu0) [inline src/resource/user/user.service.ts]: Well. our user has a unique username, but It's better if we are updating the user with their id instead.

**Review by NorrisWu0 — CHANGES_REQUESTED** [review]

File: src/resource/user/user.service.ts (hunk 1 above)
Comment (pycxu) [reply src/resource/user/user.service.ts]: Fixed: user profile is updated with `id` instead

File: src/resource/user/dto/inputs/update-profile.input.ts [hunk 2]
```diff
@@ -3,6 +3,10 @@ import { IsNotEmpty } from 'class-validator';
 
 @InputType()
 export class UpdateProfileInput {
+  @Field()
+  @IsNotEmpty()
+  id: string;
+
   @Field()
   @IsNotEmpty()
   username: string;
```
Comment (NorrisWu0) [inline src/resource/user/dto/inputs/update-profile.input.ts]: Is this username still in use? can be removed if not. Otherwise, LGTM.

**Review by NorrisWu0 — DISMISSED** [review]

File: src/resource/user/dto/inputs/update-profile.input.ts (hunk 2 above)
Comment (pycxu) [reply src/resource/user/dto/inputs/update-profile.input.ts]: Fixed

**Review by NorrisWu0 — APPROVED** [review]

**Pull request merged**